network/v1/types.go

```diff
@@ -30,6 +30,8 @@ type ClusterNetwork struct {
 	PluginName string `json:"pluginName,omitempty" protobuf:"bytes,5,opt,name=pluginName"`
 	// ClusterNetworks is a list of ClusterNetwork objects that defines the global overlay network's L3 space by specifying a set of CIDR and netmasks that the SDN can allocate addressed from.
 	ClusterNetworks []ClusterNetworkEntry `json:"clusterNetworks" protobuf:"bytes,6,rep,name=clusterNetworks"`
+	// VXLANPort is the VXLAN port used by the cluster
+	VXLANPort uint32 `json:"vxlanPort" protobuf:"varint,7,opt,name=vxlanPort"`
```

Comment:
where is the corresponding change to pkg/network/apis/network/types.go ?

Reply:
Use VXLANPort *uint32 so that we know clearly whether this optional param is set by user or not.

Reply:
@pravisankar the change to pkg/network/apis/network/types.go is in the origin PR. I agree with @danwinship on why this shouldn't be optional ([link](https://github.com/openshift/origin/pull/20024#discussion_r196233011)), is there another case that can be made? do you have an opinion @knobunc

Reply:
You can't add a new required value to an API object.  Every API field you add to a V1 field must have the implicit default as if it was not specified.  So it must be *uint32 with the default being whatever it would have been (or simply, interpreted as a default).